src/datasource.ts

```diff
@@ -338,36 +339,35 @@ export class QuickwitDataSource
     );
   }
 
-  // TODO: instead of being a string, this could be a custom type representing all the elastic types
-  // FIXME: This doesn't seem to return actual MetricFindValues, we should either change the return type
-  // or fix the implementation.
-  getFields(type?: string[], _range?: TimeRange): Observable<MetricFindValue[]> {
-    const typeMap: Record<string, string> = {
-      u64: 'number',
-      i64: 'number',
-      datetime: 'date',
-      text: 'string',
-    };
-    return from(this.getResource('indexes/' + this.index)).pipe(
-      map((index_metadata) => {
-        const shouldAddField = (field: QuickwitField) => {
-          const translated_type = typeMap[field.field_mapping.type];
+  getFields(aggregatable?: boolean, type?: string[], _range?: TimeRange): Observable<MetricFindValue[]> {
+    // TODO: use the time range.
+    return from(this.getResource('_elastic/' + this.index + '/_field_caps')).pipe(
+      map((field_capabilities_response: FieldCapabilitiesResponse) => {
+        const shouldAddField = (field: any) => {
+          if (aggregatable !== undefined && !field.aggregatable) {
```

Comment:
Shouldn't we rewrite this condition like that ? 

```javascript
if (aggregatable && !field.aggregatable)
```

Or like that ?

```javascript
if (aggregatable !== undefined && field.aggregatable != aggregatable)
```

Because like this, if I invoke this function like this `getFields(false` I'm expecting either to get all the fields that are non-aggregatable or all the fields including non-aggregatable. But like this, It'll return only the aggregatable fields which doesn't seems logic to me.

Reply:
yes indeed, the last one is better :)

Reply:
Done :)